pkg/datadog/zaplogger.go

```diff
@@ -0,0 +1,83 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package datadog // import "github.com/open-telemetry/opentelemetry-collector-contrib/pkg/datadog"
+
+import (
+	"fmt"
+
+	"go.uber.org/zap"
+)
+
+// Zaplogger implements the tracelog.Logger interface on top of a zap.Logger
+type Zaplogger struct {
```

Comment:
Do we want to add some kind of check to ensure `Zaplogger` implements `tracelog.Logger` (and `zap.Logger` too)?

Reply:
👍  added for `tracelog.Logger`
`zap.Logger` is fine, since this is not expected to implement its interface